Code/Framework/AzCore/AzCore/Task/TaskExecutor.cpp

```diff
@@ -352,11 +381,24 @@ namespace AZ
         // TODO: Something more sophisticated is likely needed here.
         // First, we are completely ignoring affinity.
         // Second, some heuristics on core availability will help distribute work more effectively
-        m_workers[++m_lastSubmission % m_threadCount].Enqueue(&task);
+        uint32_t nextWorker = ++m_lastSubmission % m_threadCount;
```

Comment:
Were you intending to change this behavior to assert when attempting to wait on a task graph from a task thread?

Reply:
Yes. At the moment any thread waiting on a task graph removes a worker from the pool. That works for single waits, but in the general case results in wedges as all workers get removed from the pool.

At this point it seems better to assert on the wait rather than have a harder to diagnose wedge.